test/common/config/utility_test.cc

```diff
@@ -200,5 +200,34 @@ TEST(UtilityTest, ObjNameLength) {
   }
 }
 
+TEST(UtilityTest, UnixCluster) {
+
+  std::string cluster_type;
+  {
+    cluster_type = "strict_dns";
+    std::string json = 
+        R"EOF({ "name": "test", "type": ")EOF" + cluster_type +
+        R"EOF(", "lb_type": "random", "connect_timeout_ms" : 1, "hosts": ["url": "unix:///test.sock"]})EOF";
+    auto json_object_ptr = Json::Factory::loadFromString(json);
+    envoy::api::v2::Cluster cluster;
+    envoy::api::v2::ConfigSource eds_config;
+    EXPECT_THROW_WITH_MESSAGE(
+        Config::CdsJson::translateCluster(*json_object_ptr, eds_config, cluster), EnvoyException,
+        "unresolved URL must be TCP scheme, got: unix:///test.sock");
+  }
+
+  {
```

Comment:
Nit: I would make these distinct tests, better to be more fine grained, in particular when there is no real boiler plate reduction in combining them (except one line).